src/common/btree.c

```diff
@@ -1756,14 +1761,21 @@ dbtree_fetch(daos_handle_t toh, dbtree_probe_opc_t opc, uint32_t intent,
 		return -DER_NO_HDL;
 
 	rc = btr_probe_key(tcx, opc, intent, key);
-	if (rc == PROBE_RC_INPROGRESS) {
+	switch (rc) {
+	case PROBE_RC_INPROGRESS:
```

Comment:
Is this even possible anymore?  The committed state of object and key trees is stored in the incarnation log, not the key/object tree.   A probe of the object or key tree will never hit uncommitted data.  Or is it a different tree?

Reply:
Here, the probe_key is not only for object/dkey/akey, but also can be for single value. If some update just refreshes the single value under an existing akey, then it is possible that some later dbtree_fetch() of such SV will hit DER_INPROGRESS.